x-pack/plugins/enterprise_search/server/lib/enterprise_search_config.ts

```diff
@@ -0,0 +1,134 @@
+});
+
+type EnterpriseSearchConfigType = TypeOf<typeof configSchema>;
+
+export const config: PluginConfigDescriptor<EnterpriseSearchConfigType> = {
+  schema: configSchema,
+  exposeToBrowser: {
+    host: true,
+  },
+};
+
+// -----------------------------------------------------------------------------
+/**
+ * Wrapper of config schema.
+ * @public
+ */
+export class EnterpriseSearchConfig {
+  /**
+   * Base URL (schema://host:port) of the Enterprise Search deployment
+   */
+  public host: string;
+
+  /**
+   * Specifies if Enterprise Search plugin should be enabled
+   */
+  public enabled: boolean;
+
+  /**
+   * Specifies the timeout for all API calls to Enterprise Search
+   */
+  public accessCheckTimeout: number;
+
+  /**
+   * Specifies a warning threshold for all API calls to Enterprise Search.
+   * If a request takes longer than this value, a user will be presented
+   * with a warning message asking them to make sure everything is OK.
+   */
+  public accessCheckTimeoutWarning: number;
+
+  /**
+   * A list of custom certificate authorities to be used for connecting to
+   * Enterprise Search APIs. If empty, default list of root CAs is used.
+   */
+  public certificateAuthorities: string[];
+
+  /**
+   * If set to true, SSL errors when connecting to Enterprise Search will be
+   * ignored as long as a successful SSL handshake could be performed.
+   */
+  public rejectUnauthorized: boolean;
+
+  /**
+   * HTTP agent object to be used for all Enterprise Search API calls
+   */
+  public httpAgent: http.Agent | https.Agent;
+
+  /**
+   * @internal
+   */
+  constructor(rawConfig: EnterpriseSearchConfigType) {
+    this.host = rawConfig.host;
+    this.enabled = rawConfig.enabled;
+    this.accessCheckTimeout = rawConfig.accessCheckTimeout;
+    this.accessCheckTimeoutWarning = rawConfig.accessCheckTimeoutWarning;
+    this.rejectUnauthorized = rawConfig.ssl.rejectUnauthorized;
+    this.certificateAuthorities = this.loadCertificateAuthorities(
+      rawConfig.ssl.certificateAuthorities
+    );
+    this.httpAgent = this.httpAgentFor(this.host);
+  }
+
+  /**
+   * Loads custom CA certificate files and returns all certificates as an array
+   **/
+  private loadCertificateAuthorities(ca) {
+    const parsedCerts = [];
+    if (ca) {
+      const paths = Array.isArray(ca) ? ca : [ca];
+      if (paths.length > 0) {
+        for (const path of paths) {
+          const parsedCert = readFileSync(path, 'utf8');
+          parsedCerts.push(parsedCert);
+        }
+      }
+    }
+    return parsedCerts;
+  }
+
+  /**
+   * Returns an HTTP agent object to be used for connecting to a given host
+   */
+  private httpAgentFor(host) {
+    try {
+      const parsedHost = new URL(host);
+      if (parsedHost.protocol === 'https:') {
+        return new https.Agent({
+          ca: this.certificateAuthorities,
+          rejectUnauthorized: this.rejectUnauthorized,
+        });
+      }
+    } catch (_error) {
+      // ignore the error
```

Comment:
nit: if you're not using the error, you can just remove the arg
```suggestion
    } catch {
      // ignore the error
```

Should we be concerned about silently swallowing the error though? Is there any chance you can describe when or where this might error that we don't care about are are OK defaulting back to a standard http agent for?

Reply:
I also think it'd be worth getting @legrego or someone else from the Security team to take a look at this file and give it a quick once over 😄

Reply:
I this case we only care about picking the right http agent type. If we can't parse the host setting from the config file, it is not the job of this particular method to care about that.

Reply:
> I also think it'd be worth getting @legrego or someone else from the Security team to take a look at this file and give it a quick once over 😄 

@jportner if you get a chance, would you mind reviewing this one on behalf of @elastic/kibana-security ? If I free up before you, then I don't mind taking a look - I just don't want to become a blocker here.